libs/scim_proto/src/lib.rs

```diff
@@ -193,4 +194,207 @@ mod tests {
         let s = serde_json::to_string_pretty(&u).expect("Failed to serialise RFC7643_USER");
         eprintln!("{}", s);
     }
+
+
+
+    // =========================================================
```

Comment:
In this case for attributes I'd waaaaaaaaaaaaaay prefer to do the from/into string handlers from `serde` over manually implementing serde; as we've only got one "custom" attribute type, we don't have to try and catch multiples with the custom serde's, and the "fallback" is much simpler. We can also use it for other things like display etc.

Reply:
Yeah, this was me just experimenting to demonstrate possible approaches. 

The other thing is that there is a difference here with attributes and subattributes. We need to start to consider that too, but it can be handled in a similar way. 

(Also at some point we need to prune old/unused values).